diff --git a/GUI/QtQuick/QtQuickAdapter.cpp b/GUI/QtQuick/QtQuickAdapter.cpp
index 69a430976..eb5e510f3 100644
--- a/GUI/QtQuick/QtQuickAdapter.cpp
+++ b/GUI/QtQuick/QtQuickAdapter.cpp
@@ -148,7 +148,6 @@ QtQuickAdapter::QtQuickAdapter(const std::shared_ptr<spdlog::logger> &logger)
    , userSigner_(std::make_shared<UserTerminal>(TerminalUsers::Signer))
    , userHWW_(bs::message::UserTerminal::create(bs::message::TerminalUsers::HWWallets))
    , txTypes_({ tr("All transactions"), tr("Received"), tr("Sent"), tr("Internal") })
-   , walletPropertiesModel_(std::make_unique<qtquick_gui::WalletPropertiesVM>())
 {
    staticLogger = logger;
    addrModel_ = new QmlAddressListModel(logger, this);
@@ -161,6 +160,7 @@ QtQuickAdapter::QtQuickAdapter(const std::shared_ptr<spdlog::logger> &logger)
    hwDeviceModel_ = new HwDeviceModel(logger, this);
    walletBalances_ = new WalletBalancesModel(logger, this);
    feeSuggModel_ = new FeeSuggestionModel(logger, this);
+   walletPropertiesModel_ = std::make_unique<qtquick_gui::WalletPropertiesVM>(logger);
 }
 
 QtQuickAdapter::~QtQuickAdapter()
@@ -242,7 +242,9 @@ void QtQuickAdapter::run(int &argc, char **argv)
    qmlRegisterType<TransactionFilterModel>("terminal.models", 1, 0, "TransactionFilterModel");
    qmlRegisterType<TransactionForAddressFilterModel>("terminal.models", 1, 0, "TransactionForAddressFilterModel");
    qmlRegisterType<PendingTransactionFilterModel>("terminal.models", 1, 0, "PendingTransactionFilterModel");
-   qmlRegisterType<qtquick_gui::WalletPropertiesVM>("terminal.models", 1, 0, "WalletPropertiesVM");
+   qmlRegisterUncreatableMetaObject(qtquick_gui::WalletPropertiesVM::staticMetaObject, "terminal.models"
+      , 1, 0, "WalletPropertiesVM", tr("Error: only enums"));
+//   qmlRegisterType<qtquick_gui::WalletPropertiesVM>("terminal.models", 1, 0, "WalletPropertiesVM");
    qmlRegisterUncreatableMetaObject(Transactions::staticMetaObject, "terminal.models" 
       , 1, 0, "Transactions", tr("Error: only enums"));
    qmlRegisterUncreatableMetaObject(TxListModel::staticMetaObject, "terminal.models"
diff --git a/GUI/QtQuick/viewmodels/WalletPropertiesVM.cpp b/GUI/QtQuick/viewmodels/WalletPropertiesVM.cpp
index b22c3e21a..8704ad9f2 100644
--- a/GUI/QtQuick/viewmodels/WalletPropertiesVM.cpp
+++ b/GUI/QtQuick/viewmodels/WalletPropertiesVM.cpp
@@ -8,12 +8,16 @@
 **********************************************************************************
 
 */
+
 #include "WalletPropertiesVM.h"
 
+#include <spdlog/spdlog.h>
+
 using namespace qtquick_gui;
 
-WalletPropertiesVM::WalletPropertiesVM(QObject* parent)
-   : QObject(parent)
+WalletPropertiesVM::WalletPropertiesVM(const std::shared_ptr<spdlog::logger> & logger, QObject* parent)
+   : QObject(parent),
+     logger_(logger)
 {
 }
 
@@ -102,11 +106,13 @@ const QStringList& WalletPropertiesVM::seed() const
 
 const QString& WalletPropertiesVM::exportPath() const
 {
+   logger_->debug("[{}] exportPath_: {}", __func__, exportPath_.toStdString());
    return exportPath_;
 }
 
 void WalletPropertiesVM::setExportPath(const QString& path)
 {
+   logger_->debug("[{}] exportPath_: {}", __func__, exportPath_.toStdString());
    exportPath_ = path;
    emit pathChanged();
 }
diff --git a/GUI/QtQuick/viewmodels/WalletPropertiesVM.h b/GUI/QtQuick/viewmodels/WalletPropertiesVM.h
index 7218c22dd..fe9c51c0c 100644
--- a/GUI/QtQuick/viewmodels/WalletPropertiesVM.h
+++ b/GUI/QtQuick/viewmodels/WalletPropertiesVM.h
@@ -10,8 +10,14 @@
 */
 #pragma once
 
+#include <memory>
+
 #include <QObject>
 
+namespace spdlog {
+   class logger;
+}
+
 namespace qtquick_gui
 {
 
@@ -39,10 +45,10 @@ class WalletPropertiesVM: public QObject
    Q_PROPERTY(bool isHardware                  READ isHardware               NOTIFY changed)
    Q_PROPERTY(bool isWatchingOnly              READ isWatchingOnly           NOTIFY changed)
    Q_PROPERTY(QStringList seed                 READ seed                     NOTIFY seedChanged)
-   Q_PROPERTY(QString exportPath READ exportPath WRITE setExportPath NOTIFY pathChanged)
+   Q_PROPERTY(QString exportPath               READ exportPath               WRITE setExportPath         NOTIFY pathChanged)
 
 public:
-   WalletPropertiesVM(QObject* parent = nullptr);
+   WalletPropertiesVM(const std::shared_ptr<spdlog::logger>&, QObject* parent = nullptr);
 
    void setWalletInfo(const WalletInfo& info);
    void setWalletSeed(const std::string& walletId, const std::string& seed);
@@ -75,6 +81,7 @@ private:
    uint32_t    nbUTXOs_{ 0 };
    QStringList seed_;
    QString exportPath_;
+   std::shared_ptr<spdlog::logger>  logger_;
 };
 
 }
